capstone/src/main/java/com/google/sps/servlets/SmallOwnedRestaurantsDataServlet.java

```diff
@@ -66,28 +65,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String restaurantName = request.getParameter("businessName");
-    boolean blackOwned = false;
-    boolean smallBusiness = true;
-    // stores tags like "pizza" or "chinese"
-    // implemented as hashset for faster lookup
-    HashSet<String> tags = new HashSet<String>();
-    // because input is string regardless of actual type, we need to extract each value in list
-    // TODO: input type might change so will need to do specific character cuts
-    //      as needed e.g. if inputted as list must omit '[' and ']' chars when storing
-    String[] tagsAsStringValues = request.getParameter("tags").split(", ");
-    for (String tag : tagsAsStringValues) {
-      tags.add(tag);
-    }
+    for (String restaurantName : restaurantNames) {
+      PlaceDetails place = details.request(restaurantName);
+
+      detailedPlaces.add(place); 
 
-    Entity restaurantEntity = new Entity("Restaurant");
-    restaurantEntity.setProperty("name", restaurantName);
-    restaurantEntity.setProperty("blackOwned", blackOwned);
-    restaurantEntity.setProperty("smallBusiness", smallBusiness);
-    restaurantEntity.setProperty("tags", tags);
+      String reviews = "";
+      PlaceDetails.Review[] reviewsList = place.reviews;
+      for (PlaceDetails.Review review : reviewsList) {
+        reviews += review.text + " ";
+      } 
+      
+      Set<String> tags = details.getTags(reviews);
+      String tagsString = tags.toString();
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(restaurantEntity);
-    response.sendRedirect("/login.html");
+      String placeString = place.toString();
+
+      boolean blackOwned = true;
+      boolean smallBusiness = false;
```

Comment:
because we're now using 2 separate databases for each filter, maybe we no longer need to store which filter it is anymore? so also removing `blackOwned` and `smallBusiness` as columns of each datastore

Reply:
Actually they were being stored under the same Entity "Restaurant", but i agree with you, separating them will be better. Especially for lookups, we can just look through whichever database our focus is.